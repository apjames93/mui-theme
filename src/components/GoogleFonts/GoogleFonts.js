import React from 'react';
import { Form, MuiInput, MuiSubmit } from 'rff-wrapper';

export const GoogleFonts = () => {
  const onSubmit = async (values) => {
    const fontLink = document.createElement('link');
    fontLink.setAttribute('href', values.fontHref);
    fontLink.setAttribute('rel', 'stylesheet');
    document.getElementById('htmlHead').appendChild(fontLink);
  };
  return (
    <Form
      onSubmit={onSubmit}
    >
      <MuiInput
        name="fontHref"
        placeholder="Google Font Link"
        fullWidth
      />
      <MuiSubmit buttonText="Add Font" />
    </Form>
  );
};

export default GoogleFonts;
