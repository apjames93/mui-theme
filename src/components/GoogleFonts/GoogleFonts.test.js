import React from 'react';
import { mount } from 'enzyme';
import { JSDOM } from 'jsdom';
import { act } from 'react-dom/test-utils';
import { Form } from 'rff-wrapper';
import renderer from 'react-test-renderer';
import GoogleFonts from './GoogleFonts';

function setup() {
  const comp = mount(
    <Form onSubmit={() => true}>
      <GoogleFonts />
    </Form>,
  );
  return { comp };
}

describe('<GoogleFonts />', () => {
  it('renders GoogleFonts', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should handleSubmit and add Link tag to head', () => {
    const dom = new JSDOM();
    const { comp } = setup();
    const { createElement } = document;
    global.document = dom.window.document;
    global.window = dom.window;
    const spy = jest.spyOn(document, 'getElementById');
    const input = comp.find('input').first();
    input.simulate('change', { target: { value: 'poop' } });
    const submitBtn = comp.find('MuiSubmit').first();
    act(async () => {
      await submitBtn.simulate('submit');
    });
    expect(spy).toBeCalledTimes(1);
    document.createElement = createElement;
  });

  test('snapshot', () => {
    const tree = renderer.create(<GoogleFonts />);
    expect(tree).toMatchSnapshot();
  });
});
