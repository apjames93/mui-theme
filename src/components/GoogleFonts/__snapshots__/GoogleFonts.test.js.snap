// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleFonts /> snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    data-testid="muiinput"
    field={
      Object {
        "input": Object {
          "checked": undefined,
          "name": "fontHref",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "type": "text",
          "value": "",
        },
        "meta": Object {
          "active": false,
          "data": Object {},
          "dirty": false,
          "dirtySinceLastSubmit": false,
          "error": undefined,
          "initial": undefined,
          "invalid": false,
          "length": undefined,
          "modified": false,
          "modifiedSinceLastSubmit": false,
          "pristine": true,
          "submitError": undefined,
          "submitFailed": false,
          "submitSucceeded": false,
          "submitting": false,
          "touched": false,
          "valid": true,
          "validating": false,
          "visited": false,
        },
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      fontHref
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete=""
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="fontHref"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Google Font Link"
        required={false}
        rows=""
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-fullWidth"
    data-testid="Button"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Add Font
    </span>
  </button>
</form>
`;
